app/src/main/java/com/quran/labs/androidquran/ui/PagerActivity.java

```diff
@@ -1983,6 +1992,28 @@ private void updateAyahBookmark(
     }
   }
 
+  private void promptForMultipleChoicePlay(int page, int startSura, int startAyah, List<Integer> startingSuraList) {
+    ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.select_dialog_item);
+    for (Integer integer : startingSuraList) {
+      String suraName = quranInfo.getSuraName(this, integer, false);
+      adapter.add(suraName);
+    }
+    if(startSura != startingSuraList.get(0)) {
+      adapter.insert("Beginning of Page", 0);
+      startingSuraList.add(0, startSura);
+    }
+    new AlertDialog.Builder(this)
```

Comment:
let's keep a reference to the dialog which we null out `onClick`, and explicitly `.dismiss()` if it's set in `onDestroy`

Reply:
I use the existing variable promptDialog if it's okay